drafts/build_a_dao_on_fuel/draft.md

```diff
@@ -0,0 +1,1095 @@
+
+## Setting up your Environment
+
+In the next few steps, you will set up your development environment to get started with building your smart contract and the front end of your Dao.
+
+1. Go to your terminal and run the command `wsl` to start your `wsl` terminal
+
+2. Next run the command to generate your new contract directory.
+
+```bash
+mkdir Fuel_Dao
+cd Fuel_Dao
+forc new contract
+```
+
+
+## Writing your Smart Contract
+
+1. Head to the `main.sw` with the command `cd main.sw` contract file, where you will be implementing your main contracts. Following the next few steps, you will be replacing the code therein.
+
+2. First you define the file as a contract file, and import other contract libraries and interfaces from the following `crates`.
+
+    - `error` - define your error handlers for your contract.
+    - `data_structre` - defines functionalities for managing proposals, voting, and defines methods for initializing your DAO.
+    - `events` - defines a library of multiple `structs` to **emit events** when certain instances, or functions are executed on-chain.
+
+    - `utils` - this will instances that will be emitted on-chain when certain events are met.
+    - `interface` - this is where you will implement you contract ABIs
+
+    All of which will require you to create other types of sway contracts, hold that thought 👍.
+
+    Copy and replace the existing code in your `main.sw` file with the one below.
+
+[note]: <> (I use Rust for all snippets rather than sway since both languages are similar and Rust's is recognized on prettifier for markdown code snippets)
+    
+```rust 
+    contract;
+
+    mod data_structures;
+    mod errors;
+    mod events;
+    mod interface;
+    mod utils;
+```
+
+3.  Next, you will import the following modules from the sway standard library; 
+    - `asset::transfer` - This provides the functionality for transferring asset
+    - `auth::mes_sender` - The module retrieves the address of the sender of the contract or a transaction.
+    - `block::height` - The module provides information about the current block height or block number
+    - `call_frames::msg_asset_id`- This provides information about the asset (token) associated with the current sent transaction.
+    - `context` - This module provides context-related functionality, related to the execution context of your contract.
+    - `hash::Hash` - This simply provides hashing-related functionality, like integrity verification, cryptographic operations, and generating unique identifiers.
+
+    Copy and add the code below to your `main.sw` contract
+    
+```rust
+    use std::{
+        asset::transfer,
+        auth::msg_sender,
+        block::height,
+        call_frames::msg_asset_id,
+        context::{
+            msg_amount,
+            this_balance,
+        },
+        hash::Hash,
+    };
+```
+
+4. Next, you will initialize a set of persisting storage that is needed throughout your contract.
```

Comment:
It's important to give context on how the storage is handled in Sway.

Reply:
I've added a brief paragraph to explain storage handling in sway and links to helpful resources for in-depth reading.